### ISSUE TITLE
USER ISSUE: Lost Items in Cancel

### ISSUE THREAD

**Issue by AliaAnor** [StrangeLoopGames/EcoIssues#18219]
\
**Version:** 0.9.0.1 beta

**Steps to Reproduce:**
1. Start a job on the butchery table with an upgrade. I had a 25% upgrade and asked for 4 small animal processing (requiring 3 carcasses).
2. Feed it 2 small animals
3. Take the upgrade out
4. Feed it the 3rd animal
5. Immediately cancel the job
5. Put the upgrade back in

**Expected behavior:**
Not entirely sure what should happen. I forgot to put the upgrade back in before having more added. What I would like is a better idea of what will happen. I was definitely trying to be tricksy with the upgrades, but would like to know the bounds rather than experiment.

**Actual behavior:**
No materials are returned.

**Do you have mods installed? Does the issue happen when no mods are installed?:**
No.

**Please provide log files. Instruction is here https://github.com/StrangeLoopGames/EcoIssues/wiki/Getting-the-Eco-game-log-file

**Comment by IcedForge** [comment]
Not getting a solid reproduction case, did you input labor and was letting this process? And  was it mid cycle on the 3rd animal carcass? Did you have any linked inventories or stockpiles; or was this only hand feeding from your inventory?

**Comment by AliaAnor** [comment]
Yes, I had inputted labor and it had begun production on the 3rd carcass. I'm not sure if it was on item 3 or 4 production. I had linked inventories. I don't want to try another repro since that involves losing some stock on my game.
